nemo/collections/asr/models/hybrid_asr_tts_models.py

```diff
@@ -0,0 +1,459 @@
+        OmegaConf.set_struct(self.cfg, True)
+
+    @classmethod
+    def from_asr_config(
+        cls,
+        asr_cfg: DictConfig,
+        asr_model_type: Union[str, ASRModelTypes],
+        tts_model_path: Union[str, Path],
+        trainer: Trainer = None,
+    ):
+        """
+        Method to construct model from ASR config for training from scratch
+        """
+        model_type = cls.ASRModelTypes(asr_model_type)
+        cfg = DictConfig(
+            dict(
+                asr_model_path=None,
+                asr_model=None,
+                tts_model_path=f"{tts_model_path}",
+                tts_model=None,
+                enhancer_model_path=None,
+                enhancer_model=None,
+                asr_model_type=f"{model_type}",
+                asr_model_fuse_bn=False,  # for training from scratch always should be False
+                train_ds=None,
+                validation_ds=None,
+                test_ds=None,
+                optim=None,
+            )
+        )
+
+        asr_cfg = copy.deepcopy(asr_cfg)  # copy not to avoid original config
+        OmegaConf.set_struct(asr_cfg, False)
+        for subconfig_path in ["optim", "train_ds", "validation_ds", "test_ds"]:
+            if subconfig_path in asr_cfg:
+                cfg[subconfig_path] = asr_cfg.pop(subconfig_path)
+        cfg.asr_model = asr_cfg
+        return cls(cfg=cfg, trainer=trainer)
+
+    @classmethod
+    def from_pretrained_models(
+        cls,
+        asr_model_path: Union[str, Path],
+        tts_model_path: Union[str, Path],
+        asr_model_fuse_bn: bool = False,
+        cfg: Optional[DictConfig] = None,
+        trainer: Optional[Trainer] = None,
+    ):
+        if cfg is None:
+            cfg = DictConfig(
+                dict(
+                    asr_model_path=f"{asr_model_path}",
+                    asr_model=None,
+                    tts_model_path=f"{tts_model_path}",
+                    tts_model=None,
+                    enhancer_model_path=None,
+                    enhancer_model=None,
+                    asr_model_type=None,
+                    asr_model_fuse_bn=asr_model_fuse_bn,
+                    train_ds=None,
+                    validation_ds=None,
+                    test_ds=None,
+                    optim=None,
+                )
+            )
+        else:
+            cfg.tts_model_path = f"{tts_model_path}"
+            cfg.asr_model_path = f"{asr_model_path}"
+        return ASRWithTTSModel(cfg, trainer=trainer)
+
+    def __setattr__(self, name, value):
+        # pytorch-lightning magic, allows to call *_step on asr_model
+        if name == "_current_fx_name" and self._full_init_guard:
+            self.asr_model._current_fx_name = value  # need to make logging inside asr_model work
+        return super().__setattr__(name, value)
+
+    def setup_optimization(
+        self, optim_config: Optional[Union[DictConfig, Dict]] = None, optim_kwargs: Optional[Dict[str, Any]] = None,
+    ):
```

Comment:
Please add docstring here why we override this method (to freeze tts model, duh! but anyway).

Reply:
Added docstring